camayoc/qcs_models.py

```diff
@@ -93,50 +213,74 @@ def __eq__(self, other):
 
 
 class NetworkProfile(QCSObject):
-    """A class to aid in CRUD test cases for network profiles."""
+    """A class to aid in CRUD test cases for network profiles.
+
+    Network profiles can be created on the quipucords server by
+    instantiating a NetworkProfile object. A unique name and username are
+    provided by default. In order to create a valid network profile,
+    you must specify at least one existing host credential and one host.
+
+    Example::
+        >>> from camayoc.qcs_models import NetworkProfile
+        >>>
+        >>> hostcred = HostCredential(password='foo')
+        >>> hostcred.create()
+        >>> netprof = NetworkProfile(hosts=['0.0.0.0'],
+                                     credential_ids=[hostcred._id])
+        >>> netprof.create()
+        >>> actual_prof = netprof.read_self().json()
+        >>>
+        >>> # the __eq__() method is customized to
+        >>> # allow == comparison between the json returned by the server
+        >>> # and the NetworkProfile object used to create it.
+        >>>
+        >>> assert actual_prof == netprof
+    """
 
     def __init__(
             self,
+            client=None,
             name=None,
             hosts=None,
             ssh_port=None,
             credential_ids=None,
             _id=None):
-        """Objects of this type contain data associated with network profiles.
-
-        Network profiles can be created on the quipucords server by
-        instantiating a NetworkProfile object. A unique name and username are
-        provided by default. In order to create a valid network profile,
-        you must specify at least one existing host credential and one host.
+        """Iniitalize a NetworkProfile object with given data.
 
-        Example::
-            >>> from camayoc import api
-            >>> from camayoc.qcs_models import NetworkProfile
-            >>> client = api.QCSClient()
-
-            >>> hostcred = HostCredential(password='foo')
-            >>> create_response = client.create_host_cred(hostcred.payload())
-            >>> hostcred._id = create_response.json()['id']
-
-            >>> netprof = NetworkProfile(hosts=['0.0.0.0'],
-                                         credential_ids=['host._id'])
-            >>> client.create_net_prof(netprof.payload())
+        If no ssh_port is supplied, it will be set to 22 by default.
+        A uuid4 name and api.Client are also supplied if none are provided.
         """
-        super().__init__(name=name, _id=_id)
+        super().__init__(client=client, name=name, _id=_id)
+        self.endpoint = QCS_PROFILES_PATH
         self.hosts = hosts
-        self.ssh_port = 22 if not name else name
+        self.ssh_port = 22 if not ssh_port else ssh_port
```

Comment:
check `if  ssh_port is not None` here, to allow sending empty string to the server. This will be useful for ensuring the validation is happening